### PR TITLE
Data types: Adding codebyte for data-type.md

### PR DESCRIPTION
**Pull request by kamlesh-ops**
### Description

- Added new sections for Number, BigInt, String, Boolean, Null, Undefined and Objects.
- closes issue #4134 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @kamlesh-ops, you've done excellent work but you have to add code bytes under ` ```codebyte/javascript`. You have added the codes as the normal examples.

**Comment by kamlesh-ops** [comment]
@avdhoottt I see. My bad. I will make another PR in few hours after rectifying.

**Comment by avdhoottt** [comment]
@kamlesh-ops You don't need to create another PR, you can just update this one. You have used ` ```js` instead of ` ```codebyte/javascript`. Just make these changes.

**Comment by avdhoottt** [comment]
We use `shell` block for the output.

**Comment by kamlesh-ops** [comment]
Sure got it

**Comment by avdhoottt** [comment]
Another thing @kamlesh-ops, have you by any chance removed or updated the existing ` ```js` codes?

**Comment by kamlesh-ops** [comment]
No I didn't modify anything, just replaced the ```js label. Are there any more issues??

**Comment by avdhoottt** [comment]
@kamlesh-ops Nope, you shouldn't have done that. You must create a new code byte at the end of the existing content. Make the entry as it was and add the code byte as instructed. You can refer to our [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md),

**Comment by kamlesh-ops** [comment]
@avdhoottt Sir I have made some changes. Actually, I had restructured the existing code and explanations, so I have just added the new changed section under the previous section, you may find two sections for string, for example, the new modified one and the old one. 
Pl verify and let me know. 
Thanks.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @kamlesh-ops, I appreciate your work but I need to tell you that there is a fixed structure for each entry as follows - 
- Heading
- One line description
- Normal Code block
- Output of the above code block
- Then the code byte.

Please follow these. Then it would help if you made these - 
- You have added 2 String headings.
- You don't need to add concepts like string interpolation, etc in this entry as this entry should only give a basic overview of the data types.
- You have two boolean headings, keep only one.
- The same goes for Null and Undefined headings.

I suggest keeping the previous codes (those were written before you made this PR) and then adding the codebytes only below the old content. You can edit the old content too but try to keep it as simple as possible. It's better to keep one codebyte for one heading if possible.

Thanks.

**Comment by kamlesh-ops** [comment]
@avdhoottt I get the idea. Actually this is my first PR, so maybe that's why the mistake. I will resolve the problem and get back.

**Comment by kamlesh-ops** [comment]
@avdhoottt I hope I have made the changes as you had instructed.

File: content/javascript/concepts/data-types/data-types.md [hunk 1]
```diff
@@ -36,7 +36,8 @@ JavaScript uses seven primitive data types that are listed in the table below:
 
 > **Note:** The `null` type cannot be tested with `typeof` like all the other primitive types because it will return `"object"`. This must be tested with the type equality operator (i.e., `=== null`).
 
-## Number and BigInt
+## Number and BigInt  
+In JavaScript, numbers are always stored as double-precision floating point numbers.
```
Comment (avdhoottt) [inline content/javascript/concepts/data-types/data-types.md:40]:
```suggestion

In JavaScript, numbers are always stored as double-precision floating point numbers.
```

File: content/javascript/concepts/data-types/data-types.md [hunk 2]
```diff
@@ -55,8 +57,17 @@ This will output the following:
 ```shell
 10000000000000000
 9999999999999999n
-```
+```  
+
+The integer precision for the `Number` type is `15 digits`. You can see for yourself.
```
Comment (avdhoottt) [inline content/javascript/concepts/data-types/data-types.md:62]:
```suggestion
The integer precision for the `Number` type is `15 digits`. The following example is runnable: 
```

File: content/javascript/concepts/data-types/data-types.md [hunk 3]
```diff
@@ -75,6 +93,13 @@ let lateToWork = true;
 let isTheEarthFlat = false;
 ```
 
+To convert a non-boolean value into a boolean, the `Boolean` function or double not `!!` is used.
+
+```codebyte/javascript
+let value = Boolean('Harry'); 
+console.log(value); // returns true
```
Comment (avdhoottt) [inline content/javascript/concepts/data-types/data-types.md:100]:
```suggestion
let value = Boolean('Harry'); 
console.log(value);
```

**Review by avdhoottt — APPROVED** [review]
Excellent work @kamlesh-ops 🚀, it's good to go for the second review.

**Comment by kamlesh-ops** [comment]
@avdhoottt great to know! Thanks:)

File: content/javascript/concepts/data-types/data-types.md [hunk 4]
```diff
@@ -66,6 +77,13 @@ let greeting = 'Hi buddy';
 let message = 'You are doing great! Keep studying!';
 ```
 
+To find the length of the string, the built-in `length` property is used.
```
Comment (ishg-153) [inline content/javascript/concepts/data-types/data-types.md:80]:
```suggestion
To find the length of the string, the built-in `length` property is used. Run the following example to get a better idea:
```

File: content/javascript/concepts/data-types/data-types.md [hunk 5]
```diff
@@ -75,6 +93,13 @@ let lateToWork = true;
 let isTheEarthFlat = false;
 ```
 
+To convert a non-boolean value into a boolean, the `Boolean` function or double not `!!` is used.
```
Comment (ishg-153) [inline content/javascript/concepts/data-types/data-types.md:96]:
```suggestion
To convert a non-boolean value into a boolean, the `Boolean` function or double not `!!` is used. Run the following code block to see how `Boolean()` works:
```

File: content/javascript/concepts/data-types/data-types.md [hunk 6]
```diff
@@ -98,3 +141,15 @@ const user = {
   isActive: true,
 };
 ```
+
+Alternately, an empty `object` can be initialised and then the properties added to the object.
```
Comment (ishg-153) [inline content/javascript/concepts/data-types/data-types.md:145]:
```suggestion
Alternately, an empty `object` can be initialized and the properties added to the object. Run the code below to see it in action:
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@kamlesh-ops I've requested a few changes please make those.

**Comment by kamlesh-ops** [comment]
@ishg-153 made the changes as requested.

**Review by ishg-153 — APPROVED** [review]
@kamlesh-ops looks good to merge to me.

**Pull request merged**